.github/workflows/ci.yml

```diff
@@ -97,30 +118,21 @@ jobs:
           if [ "${{ matrix.backend }}" = "vulkan" ]; then
             brew install molten-vk vulkan-headers
           fi
-      # Regular tests
-      - run: just ci-test ${{ matrix.backend }}
-      - run: just test-doc ${{ matrix.backend }}
-      # Test MSRV
+      # Setup rust version
+      - run: rustup update stable && rustup default stable
+        if: matrix.rust-version == 'stable'
       - name: Read MSRV
         id: msrv
         run: echo "value=$(sed -nE 's/^rust-version[[:space:]]*=[[:space:]]*"([^"]+)".*/\1/p' Cargo.toml)" >> $GITHUB_OUTPUT
+        if: matrix.rust-version == 'msrv'
       - name: Install Rust
         uses: dtolnay/rust-toolchain@stable
         with:
           toolchain: ${{ steps.msrv.outputs.value }}
+        if: matrix.rust-version == 'msrv'
+      # Run tests
       - run: just ci-test ${{ matrix.backend }}
-      # Test MIRI
-      # Currently disabled due to "unsupported operation: can't call foreign function `mln$bridge$cxxbridge1$MapRenderer_new` on OS `linux`"
-      # Needs reinvestigation if this can be fixed via "-Zmiri-native-lib" or maybe "-Zmiri-native-lib-enable-tracing"
-      # Tracking issue upstream https://github.com/rust-lang/miri/issues/11
-      # - run: |
-      #     echo "NIGHTLY=nightly-$(curl -s https://rust-lang.github.io/rustup-components-history/x86_64-unknown-linux-gnu/miri)" >> "$GITHUB_ENV"
-      # - name: Install ${{ env.NIGHTLY }}
-      #   uses: dtolnay/rust-toolchain@master
-      #   with:
-      #     toolchain: ${{ env.NIGHTLY }}
-      #     components: miri
-      # - run: just test-miri ${{ matrix.backend }}
+      - run: just test-doc ${{ matrix.backend }}
```

Comment:
lets add `test-doc` into `ci-test` - the whole idea of `ci-test` is to combine all steps we want to do in CI